src/errors.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { GitErrorCodes } from "./git";
+import { localize } from "./utils/localize";
+
+export class NoWorkspaceError extends Error {
+    public message: string = localize('noWorkspaceError', 'You must have a workspace open to perform this operation.');
+}
+
+// copied from https://github.com/microsoft/vscode/blob/main/extensions/git/src/commands.ts#L2800-L2863
```

Comment:
nit: I think it would be good to use a permalink for this URL in case the line numbers change: https://github.com/microsoft/vscode/blob/779434d2d118889e2a5a2113714ad6c8bcb3a6e3/extensions/git/src/commands.ts#L2800-L2863

Reply:
```suggestion
// copied from https://github.com/microsoft/vscode/blob/779434d2d118889e2a5a2113714ad6c8bcb3a6e3/extensions/git/src/commands.ts#L2800-L2863
```